lib/Table/TableStyle.ts

```diff
@@ -471,29 +468,52 @@ export default class TableStyle {
     } else {
       // No column to color by, so use the same color for everything.
       let color: Color | undefined;
+      // May use nullColor depending on colorColumn (eg if of type 'region')
+      let nullColor: Color | undefined;
       const colorId = this.tableModel.uniqueId || this.tableModel.name;
-      if (colorTraits.nullColor) {
+
+      // If colorColumn is of type region - use regionColor
+      if (colorColumn?.type === TableColumnType.region) {
+        color = this.regionColor;
+        nullColor = this.nullColor;
+      } else if (colorTraits.nullColor) {
         color = Color.fromCssColorString(colorTraits.nullColor);
       } else if (this.binColors.length > 0) {
         color = this.binColors[0];
       } else if (colorId) {
         color = Color.fromCssColorString(getColorForId(colorId));
       }
 
-      return new ConstantColorMap(
-        color ?? Color.fromCssColorString(defaultColor),
-        this.tableModel.name
-      );
+      if (!color) {
+        color = this.defaultColor;
+      }
+
+      return new ConstantColorMap({
+        color,
+        title: this.tableModel.name,
+        nullColor
+      });
     }
   }
 
+  @computed get defaultColor() {
+    return Color.fromCssColorString(defaultColor);
+  }
+
```

Comment:
This shouldn't be a computed because it doesn't use anything observable. Does it even need to be in this class?

Reply:
Sorry I forgot to remove that - done!